model/convert/service_event.go

```diff
@@ -263,16 +273,25 @@ func convertServiceEventEpochCommit(event flow.Event) (*flow.ServiceEvent, error
 	}
 
 	// parse DKG group key and participants
-	// Note: this is read in the same order as `DKGClient.SubmitResult` ie. with the group public key first followed by individual keys
-	// https://github.com/onflow/flow-go/blob/feature/dkg/module/dkg/client.go#L182-L183
-	dkgGroupKey, dkgParticipantKeys, err := convertDKGKeys(cdcDKGKeys.Values)
+	commit.DKGParticipantKeys, err = convertDKGKeys(cdcDKGKeys.Values...)
 	if err != nil {
 		return nil, fmt.Errorf("could not convert DKG keys: %w", err)
 	}
-	commit.DKGGroupKey = dkgGroupKey
-	commit.DKGParticipantKeys = dkgParticipantKeys
-	// TODO(EFM, #6214): parse index map from service event
-	commit.DKGIndexMap = nil
+	groupKey, err := convertDKGKeys(cdcDKGGroupKey)
+	if err != nil {
+		return nil, fmt.Errorf("could not convert DKG group key: %w", err)
+	}
+	commit.DKGGroupKey = groupKey[0]
```

Comment:
I find constructions like that less clear, where we implicitly know the length of the slice. 

We could break up `convertDKGKeys` into an implementation that converts a single key (e.g. `convertDKGKey`), and a second function that just iterates over a slice and calls `convertDKGKey`. I think that would be more clear.

Reply:
I agree with you, liked your other suggestion a lot. [`0e9e8d8` (#6494)](https://github.com/onflow/flow-go/pull/6494/commits/0e9e8d88fef64bcf858ab7a609be2eeb07cc601b)